From 3a71c9bc3747201e5bebe0e80b98ac6219209875 Mon Sep 17 00:00:00 2001
From: alperak <alperyasinak1@gmail.com>
Date: Thu, 8 Feb 2024 14:09:32 +0300
Subject: [PATCH] Patch versioneer for Python 3.12 compatibility

AttributeError: 'ConfigParser' object has no attribute 'readfp'. Did you mean: 'read'?
AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?

readfp and SafeConfigParser has been deprecated since Python 3.2 and removed in Python 3.12 (due October 2023). Use read_file and ConfigParser instead.

https://docs.python.org/3/whatsnew/3.12.html#configparser

Upstream-Status: Submitted [https://github.com/irgeek/StrEnum/pull/34]
Signed-off-by: alperak <alperyasinak1@gmail.com>
---
 versioneer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/versioneer.py b/versioneer.py
index 64fea1c..3aa5da3 100644
--- a/versioneer.py
+++ b/versioneer.py
@@ -339,9 +339,9 @@ def get_config_from_root(root):
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
-        parser.readfp(f)
+        parser.read_file(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory

     def get(parser, name):
--
2.25.1

